test/functional/rpc_packages.py

```diff
@@ -91,52 +68,65 @@ def run_test(self):
     def test_independent(self):
         self.log.info("Test multiple independent transactions in a package")
         node = self.nodes[0]
+
         # For independent transactions, order doesn't matter.
         self.assert_testres_equal(self.independent_txns_hex, self.independent_txns_testres)
 
         self.log.info("Test an otherwise valid package with an extra garbage tx appended")
-        garbage_tx = node.createrawtransaction([{"txid": "00" * 32, "vout": 5}], {self.address: 1})
-        tx = tx_from_hex(garbage_tx)
+        garbage_utxo = {"txid": "00" * 32, "vout": 5, "value": 3, "height": 0}
+        garbage_tx = self.wallet.create_self_transfer(utxo_to_spend=garbage_utxo)
         # Only the txid and wtxids are returned because validation is incomplete for the independent txns.
         # Package validation is atomic: if the node cannot find a UTXO for any single tx in the package,
         # it terminates immediately to avoid unnecessary, expensive signature verification.
-        package_bad = self.independent_txns_hex + [garbage_tx]
-        testres_bad = self.independent_txns_testres_blank + [{"txid": tx.rehash(), "wtxid": tx.getwtxid(), "allowed": False, "reject-reason": "missing-inputs"}]
+        package_bad = self.independent_txns_hex + [garbage_tx["hex"]]
+        testres_bad = self.independent_txns_testres_blank + [{"txid": garbage_tx["txid"], "wtxid": garbage_tx["wtxid"], "allowed": False, "reject-reason": "missing-inputs"}]
         self.assert_testres_equal(package_bad, testres_bad)
 
         self.log.info("Check testmempoolaccept tells us when some transactions completed validation successfully")
-        coin = self.coins.pop()
-        tx_bad_sig_hex = node.createrawtransaction([{"txid": coin["txid"], "vout": 0}],
-                                           {self.address : coin["amount"] - Decimal("0.0001")})
-        tx_bad_sig = tx_from_hex(tx_bad_sig_hex)
+        # get an UTXO that requires signature to be spent
+        deterministic_address = node.get_deterministic_priv_key().address
+        blockhash = self.generatetoaddress(node, 1, deterministic_address)[0]
+        coinbase = node.getblock(blockhash=blockhash, verbosity=2)["tx"][0]
+        coin = {
+                "txid": coinbase["txid"],
+                "value": coinbase["vout"][0]["value"],
+                "scriptPubKey": coinbase["vout"][0]["scriptPubKey"],
+                "vout": 0,
+                "height": 0
+            }
+        self.generate(node, COINBASE_MATURITY, sync_fun=self.no_op)
```

Comment:
Maybe generate all the coins at the beginning so we don't need to multiple chains of 100 blocks for coinbase maturity?

Reply:
Done in b21329d50639fc383d92c75ab7f20187de4ae0e9.